doc/benchmark.md

```diff
@@ -0,0 +1,1328 @@
+      del:
+      set:
+      add:
+    body: >+
+      <!DOCTYPE html>
+      <html>
+      <head>
+      <title>Welcome to easegress!</title>
+      <style>
+          body {
+            width: 35em;
+            margin: 0 auto;
+            font-family: Tahoma, Verdana, Arial, sans-serif;
+          }
+      </style>
+      </head>
+      <body>
+      <h1>Welcome to easegress!</h1>
+      <p>If you see this page, the easegress web server is successfully installed and working. Further configuration is required.</p>
+
+      <p>For online documentation and support please refer to <a href="http://easegress.org/">easegress.org</a>.<br/> Commercial support is available at <a href="http://easegress.com/">easegress.com</a>.</p>
+      <p><em>Thank you for using easegress.</em></p>
+      </body>
+      </html>
+```
+
+**Traefik**
+1. Running binary directly with command `./traefik -c ./traefik.yml` [1]
+2. Static lonfig: traefik.yaml
+``` yaml 
+log:
+  level: INFO
+
+entryPoints:
+  web:
+    address: ":8081"
+
+providers:
+  file:
+    filename: /${file_path}/dynamic_conf.yml
+
+```
+3. Dynamic config: dynamic_conf.yml
+``` yaml
+http:
+  routers:
+    my-router:
+      rule: "PathPrefix(`/pipeline`)"
+      service: foo
+      entryPoints:
+      - web
+
+  services:
+    foo:
+      loadBalancer:
+        servers:
+        - url: "http://${echo_server_ip}:9095"
+```
+
+**Echo Server**
+1. [Source code](https://github.com/megaease/easegress/tree/main/example/backend-service/mirror)
+2. Its logic is accepting HTTP request and printing to the console, in this testing, we only uses `9095` port.
+
+
+## Testing 
+### Baseline test
+1. **Echo-server**
+* Loading echo-server directly from `vm02` to `vm03`
+* **Scenario 1**: 50 concurrency/900 requests/2 miniutes limitation/not QPS limitation
+
+``` bash
+
+./hey -n 900   -c 50  -m GET http://${vm03_ip}:9095/pipeline -z 2m    
+
+```
+
+* **Scenario 2**: 100 concurrency/90000 requests/2 miniutes limitation/not QPS limitation
+``` bash
+
+./hey -n 90000  -c 100  -m GET http://${vm03_ip}:9095/pipeline -z 2m   
+
+```
+
+* **Scenario 3**: 120 concurrency/90000 requests/ 2 miniutes limitation/not QPS limitation
+
+``` bash
+
+./hey -n 90000  -c 120  -m GET http://${vm03_ip}:9095/pipeline -z 2m   
+
+```
+
+| Scenario | Total | Slowest | Fastest | Average | RPS   | 90% Latency | 95% Latency | 99% Latency | load average(top -c ) |
+| -------- | ----- | ------- | ------- | ------- | ----- | ----------- | ----------- | ----------- | --------------------- |
+| #1       | 0.19s | 0.015s  | 0.0103s | 0.0109s | 4517  | 0.0119s     | 0.0126s     | 0.0144s     | 0/0/0                 |
+| #2       | 9.88s | 0.054s  | 0.0101s | 0.0109s | 9109  | 0.0118s     | 0.0124s     | 0.0138s     | 0.96/0.42/0.19        |
+| #3       | 8.48s | 0.042s  | 0.0101s | 0.0110s | 10768 | 0.0122s     | 0.0129s     | 0.0149s     | 1.34/0.46/0.20        |
+
+
+
+2. **Nginx**
+* Loading Nginx's `index.html` url, from `vm02` to `vm01`.
```

Comment:
There are `Nginx`, `NGINX`, and `nginx` in the doc, propose to use the same style.

Reply:
OK, I will update `Nginx` to `NGINX`. Since some `nginx` format usages are unavoidable, e.g., `pid /run/nginx.pid;`, I will keep them as `nginx`.